packages/editor/src/components/parts/common/customfield/CustomFieldTable.tsx

```diff
@@ -1,43 +1,55 @@
-import type { WfCustomField, CustomFieldConfigType } from '@axonivy/inscription-protocol';
+import type { WfCustomField, WorkflowType } from '@axonivy/inscription-protocol';
 import { CUSTOM_FIELD_TYPE } from '@axonivy/inscription-protocol';
 import { IvyIcons } from '@axonivy/editor-icons';
 import type { ColumnDef } from '@tanstack/react-table';
 import { flexRender } from '@tanstack/react-table';
 import { memo, useMemo } from 'react';
 import type { SelectItem } from '../../../../components/widgets';
 import {
-  EditableCell,
   SelectCell,
   Table,
   TableHeader,
   TableCell,
   TableAddRow,
   SortableHeader,
   ScriptCell,
-  ResizableHeader
+  ResizableHeader,
+  ComboCell
 } from '../../../../components/widgets';
-import { useAction } from '../../../../context';
+import { useAction, useEditorContext, useMeta, usePath } from '../../../../context';
 import { SelectableValidationRow } from '../path/validation/ValidationRow';
 import { PathCollapsible } from '../path/PathCollapsible';
 import { useResizableEditableTable } from '../table/useResizableEditableTable';
+import { toWorkflowType } from '../info/Information';
+import { deepEqual } from '../../../../utils/equals';
 
 type CustomFieldTableProps = {
   data: WfCustomField[];
   onChange: (change: WfCustomField[]) => void;
-  type: CustomFieldConfigType;
+  type: WorkflowType;
 };
 
 const EMPTY_WFCUSTOMFIELD: WfCustomField = { name: '', type: 'STRING', value: '' } as const;
 
 const CustomFieldTable = ({ data, onChange, type }: CustomFieldTableProps) => {
   const items = useMemo<SelectItem[]>(() => Object.entries(CUSTOM_FIELD_TYPE).map(([value, label]) => ({ label, value })), []);
 
+  const { context } = useEditorContext();
+  const path = usePath();
+
+  const predefinedCustomField: WfCustomField[] = useMeta('meta/workflow/customFields', { context, type: toWorkflowType(path) }, []).data;
```

Comment:
toWorkflowType IMO not needed, type is already there from the component props

Reply:
Ah now I get it, thanks. I will remove the toWorkflowType method from CustomFieldTable 👍